### PR TITLE
Pr finalize v1alpha2

### PR DESCRIPTION
**Pull request by sbueringer**
Final fixes for v1alpha2. This has been tested with:
* [x] multi-node cluster with LBaaS
* [x] single-nocde cluster without LBaaS

### CODE REVIEW

**Comment by sbueringer** [comment]
/assign @jichenjc 
/assign @hidekazuna 

After this only the staging release job and docs should be left for v1alpha2

File: examples/generate.sh [hunk 1]
```diff
@@ -144,24 +145,34 @@ if [[ "$DOMAIN_NAME" = "null" ]]; then
...
   OPENSTACK_CLOUD_PROVIDER_CONF="$OPENSTACK_CLOUD_PROVIDER_CONF
-          region=\"${REGION}\"
+region=\"${REGION}\"
   "
 fi
+
+export OPENSTACK_CLOUD_PROVIDER_CONF6
+OPENSTACK_CLOUD_PROVIDER_CONF6=$(echo "${OPENSTACK_CLOUD_PROVIDER_CONF}" | sed -e 's/^/      /')
+export OPENSTACK_CLOUD_PROVIDER_CONF10
+OPENSTACK_CLOUD_PROVIDER_CONF10=$(echo "${OPENSTACK_CLOUD_PROVIDER_CONF}" | sed -e 's/^/          /')
```
Comment (jichenjc) [inline examples/generate.sh]: I didn't get this CONF6/CONF10 ... what's the meaning of this?

File: examples/generate.sh [hunk 2]
```diff
@@ -199,25 +210,19 @@ echo "Generated ${CONTROLPLANE_GENERATED_FILE}"
...
-echo "---" >> ${MACHINES_GENERATED_FILE}
-cat ${MACHINEDEPLOYMENT_GENERATED_FILE} >> ${MACHINES_GENERATED_FILE}
-echo "---" >> ${MACHINES_GENERATED_FILE}
-echo "Generated ${MACHINES_GENERATED_FILE}"
+cp ${SOURCE_DIR}/addons.yaml "${ADDONS_GENERATED_FILE}"
+echo "Generated ${ADDONS_GENERATED_FILE}"
 
 # Generate Cluster API provider components file.
-kustomize build "github.com/kubernetes-sigs/cluster-api//config/default/?ref=master" --reorder=none > "${COMPONENTS_CLUSTER_API_GENERATED_FILE}"
+kustomize build "github.com/kubernetes-sigs/cluster-api//config/default/?ref=master" > "${COMPONENTS_CLUSTER_API_GENERATED_FILE}"
```
Comment (jichenjc) [inline examples/generate.sh]: nit: use / instead //

File: examples/generate.sh (hunk 1 above)
Comment (sbueringer) [reply examples/generate.sh]: The problem is that I sometimes need the config with indent 6 (KubeadmConfig) and sometimes with indent 10 (KubeadmConfigTemplate). Seems to be the easiest way to just have to vars for envsubst.

File: examples/generate.sh (hunk 2 above)
Comment (sbueringer) [reply examples/generate.sh]: Done

**Comment by sbueringer** [comment]
@jichenjc ptal :)

File: examples/generate.sh (hunk 1 above)
Comment (jichenjc) [reply examples/generate.sh]: ok, got it thanks

**Comment by CamelCaseNotation** [comment]
@sbueringer could you share your YAML for creating MM cluster with LBaaS, as well as single node cluster without LBaaS? It would be greatly appreciated!!

**Comment by sbueringer** [comment]
> @sbueringer could you share your YAML for creating MM cluster with LBaaS, as well as single node cluster without LBaaS? It would be greatly appreciated!!

That's the one for single node control plane:
https://gist.github.com/sbueringer/35e21417670a6a0c4983bad2911ea538

But please just use the documentation (and examples/generate.sh), which should be up-to-date with this PR: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/474

There's also an issue to generate separate single-node and multi-node control plane examples: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/473

I also have my dev-env published here: https://github.com/sbueringer/capi-dev/tree/master/cluster

Which first generates examples with the scripts from the repo and then kustomizes them to my env. If you have further questions you can always ask :)

**Pull request merged**